tests/phpunit/integration/Core/Notifications/NotificationsTest.php

```diff
@@ -42,4 +47,35 @@ function ( REST_Route $route ) {
 			$routes
 		);
 	}
+
+	public function test_non_proxy_request_does_not_request_proxy() {
+		// Create a user with access to the WP REST API and log in.
+		$user = $this->factory()->user->create_and_get( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user->ID );
+		do_action( 'wp_login', $user->user_login, $user );
+
+		// Fake setup and authentication for access to dashboard.
+		$this->fake_site_connection();
+		remove_all_filters( 'googlesitekit_setup_complete' );
+		$authentication = new Authentication( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$authentication->verification()->set( true );
+		$authentication->get_oauth_client()->set_access_token( 'test-access-token', HOUR_IN_SECONDS );
+
+		// Set up the REST Routes including the notification route to test.
+		$rest_routes = new REST_Routes( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$rest_routes->register();
```

Comment:
This part isn't actually needed because the routes are set up during the plugin bootstrap. I like the idea to use dedicated instances in the tests but I'm not sure if we can. This is also still using the `Route` instance from the main/bootstrapped `Notifications` class.